src/transformers/modeling_utils.py

```diff
@@ -1407,6 +1407,8 @@ def __init__(self, config: PretrainedConfig, *inputs, **kwargs):
                 "`PretrainedConfig`. To create a model from a pretrained model use "
                 f"`model = {self.__class__.__name__}.from_pretrained(PRETRAINED_MODEL_NAME)`"
             )
+        if hasattr(config, "validate"):  # e.g. in @strict_dataclass
+            config.validate()
```

Comment:
Not done yet but yes, could be a solution for "class-wide validation" in addition to "per-attribute validation"